### PR TITLE
Fix torch.combinations RuntimeError with dynamic shapes

### LINKED ISSUE #163759: [dynamic] [dynamic] `torch.combinations` throws strange errors when `backend=aot_eager, dynamic=True`
### 🐛 Describe the bug

`torch.combinations` throws strange errors when compiled by `aot_eager` when `dynamic=True`
```python
import torch
import torch.nn as nn
import torch.nn.functional as F
from torch._inductor import config

config.fallback_random = True
torch.set_grad_enabled(False)


class Model(torch.nn.Module):

    def __init__(self):
        super(Model, self).__init__()

    def forward(self, x):
        x = torch.combinations(x.flatten(), r=2)
        return x


model = Model()


x = torch.randn(3, 1, 1)

inputs = [x]

device = "cpu"
def run_test(model, inputs, device, backend):
    torch.manual_seed(0)
    model.to(device)
    inputs = [input.to(device) for input in inputs]
    if backend != "eager":
        model = torch.compile(model, backend=backend, dynamic=True)
    try:
        output = model(*inputs)
        print(f"succeed on {backend}: {output}")
    except Exception as e:
        print(e)


run_test(model, inputs, device, 'eager')
run_test(model, inputs, device, 'aot_eager')
```

### Error logs

```
succeed on eager: tensor([[-0.9697, -0.2773],
        [-0.9697,  0.3364],
        [-0.2773,  0.3364]])
got RuntimeError('Cannot call numel() on tensor with symbolic sizes/strides\nException raised from throw_cannot_call_with_symbolic at /pytorch/c10/core/TensorImpl.cpp:291 (most recent call first):\
```

### Versions

2.10.0.dev20250923+cu126

### PR DESCRIPTION
**Pull request by salmanmkc**
The C++ implementation of torch.combinations calls numel() which fails when tensors have symbolic shapes (used in dynamic compilation with aot_eager). This adds a Python decomposition that uses size(0) instead, which properly handles symbolic shapes.

The decomposition implements the same algorithm as the C++ version:
- Creates meshgrid of r copies of the input tensor
- Creates an index-based triangular mask for valid combinations
- Filters grids using the mask and stacks the results

Fixes the issue where torch.compile(model, backend='aot_eager', dynamic=True) would fail with RuntimeError when calling torch.combinations.

Fixes #163759

cc @mlazos

### CODE REVIEW

**Comment by salmanmkc** [comment]
Closing in favor of #163831 which fixes this issue more directly by using sym_numel() in the C++ kernel implementation rather than adding a decomposition.
